agent/agent-tooling/src/main/java/com/azure/monitor/opentelemetry/exporter/implementation/localstorage/LocalStorageTelemetryPipelineListener.java

```diff
@@ -47,8 +50,12 @@ public LocalStorageTelemetryPipelineListener(
   }
 
   public void shutdown() {
-    localFileSender.shutdown();
-    localFilePurger.shutdown();
+    // guarding against multiple shutdown calls because this can get called if statsbeat shuts down
+    // early because it cannot reach breeze and later on real shut down (when running not as agent)
+    if (!shutdown.getAndSet(true)) {
+      localFileSender.shutdown();
```

Comment:
Can we add a one time log?

Reply:
it's not uncommon for shutdown to be silent when called multiple times, I think I'd prefer to leave it unless you have a specific reason why you want to log it

Reply:
we log when shutting down statsbeat scheduler.. i thought it will nice to log here when shutting down disk persistence. i think you misunderstood what i meant?

Reply:
ah, as debug?

Reply:
I went to add it but realized it could be confusing because in this class we don't know if it's statsbeat or not, and so it's not super clear what exactly to log that wouldn't cause more confusion than less. I think i'd prefer to leave it as is if that's ok.

Reply:
yeah, it's only for Statsbeat. i'm ok with it.

Reply:
i didn't test this part with 3.2.11. might need to test it out later.